### DISCUSSION TITLE
How do I handle isDirty with numbers?

### DISCUSSION THREAD

**Discussion by Kikkomann** [react-hook-form/react-hook-form#9588] [Q&A] [answered]
I have an object containing a number (year) and isDirty is not reset to false, when the initial number is entered. I know that [input should only be strings](https://github.com/react-hook-form/react-hook-form/issues/2644#issuecomment-675753217), but the [codesandbox](https://codesandbox.io/s/react-hook-form-parse-and-format-textarea-furtc) shows a workaround for using numbers. I tried it using `register` (thanks to [this](https://github.com/react-hook-form/react-hook-form/issues/4597#issuecomment-1069586878)), but it does not work for me:

```typescript
<TextField // From material UI
  label="Year"
  type="number"
  {...register("year", {
    onChange: (event) => parseInt(event.target.value, 10),
  })}
/>
```

Once I update the `year`, it is dirty. Even after going back to the initial value.

What am I missing?

**Comment by bluebill1049** [comment, marked as answer]
```tsx
import { useForm } from "react-hook-form";
import Headers from "./Header";
import "./styles.css";

export default function App() {
  const {
    register,
    handleSubmit,
    formState: { isDirty }
  } = useForm({
    defaultValues: {
      number: 1
    }
  });

  console.log("isDirty", isDirty);

  return (
    <div>
      <Headers />

      <form onSubmit={handleSubmit((data) => console.log(data))}>
        <input
          {...register("number", { valueAsNumber: true })}
          placeholder="First Name"
        />
        <input type="submit" />
      </form>
    </div>
  );
}
```
https://codesandbox.io/s/reverent-borg-qslovj?file=/src/App.tsx:0-592

does this help?
